lib/networks/network_context.go

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright (c) 2020 - present Kurtosis Technologies LLC.
+ * All Rights Reserved.
+ */
+
+package networks
+
+import (
+	"fmt"
+	"github.com/google/uuid"
+	"github.com/kurtosis-tech/kurtosis-go/lib/kurtosis_service"
+	"github.com/kurtosis-tech/kurtosis-go/lib/services"
+	"github.com/palantir/stacktrace"
+	"github.com/sirupsen/logrus"
+	"os"
+	"path/filepath"
+)
+
+const (
+	ipPlaceholder = "KURTOSISSERVICEIP"
+)
+
+/*
+The identifier used for services with the network.
+*/
+type ServiceID string
```

Comment:
We were going to rename this to `ServiceTag` and I thought about doing it now, but I actually decided against it - if we ever wanted to actually tag services (to identify sets of services) then we'd have a name conflict.